### PR TITLE
Snap packaging

### PR DESCRIPTION
**Pull request by blakerouse**
Adds snap/snapcraft.yaml and improves the help menu to use the snap instance name. Add maas-cli content interface so "maas" snap can connect to this snap to get the latest CLI.

### CODE REVIEW

File: maas/client/flesh/__init__.py [hunk 1]
```diff
@@ -511,8 +512,18 @@ def post_mortem(traceback):
     post_mortem(traceback)
 
 
+def program_name_from_env(program):
+    """Return the program name from environment."""
+    if os.environ.get("SNAP_INSTANCE_NAME"):
+        return os.environ.get("SNAP_INSTANCE_NAME")
+    if os.environ.get("SNAP_NAME"):
+        return os.environ.get("SNAP_NAME")
```
Comment (albertodonato) [inline maas/client/flesh/__init__.py]: you shouldn't need this, `SNAP_INSTANCE_NAME` is always there for the snap

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  physical servers for amazing data center operational efficiency.
+grade: stable
+confinement: strict
+
+slots:
+  maas-cli:
+    interface: content
+    content: maas-cli
+    read:
+      - $SNAP/lib
```
Comment (albertodonato) [inline snap/snapcraft.yaml]: can't we just expose the path of the library?

Comment (blakerouse) [reply snap/snapcraft.yaml]: We need all the dependencies not just the libmaas pieces, so I thought it best to expose the whole lib section. Only thing under there is lib/python3.6/site-packages anyway.

File: maas/client/flesh/__init__.py (hunk 1 above)
Comment (blakerouse) [reply maas/client/flesh/__init__.py]: Okay will remove it, didn't know if that would always be the case, if you had a older snapd.

Comment (albertodonato) [reply maas/client/flesh/__init__.py]: I think that got introduced way before content interface :)

**Pull request merged**